woodwork/schema.py

```diff
@@ -303,10 +312,11 @@ def set_index(self, new_index):
         if old_index is not None:
             self.remove_semantic_tags({old_index: 'index'})
```

Comment:
There are some validation checks that happen in `remove_semantic_tags`, but as `old_index` will always be None at `init`, we can't reach this section of `set_index` (and the corresponding section in `set_time_index`) in Schema init. 

However, we're now exposing `validate` in `set_index` and `set_time_index` so, in theory, users could pass `validate=False` while setting a Schema's index directly, and we'd miss these validation checks.

However, since the Accessor `set_index` and `set_time_index` methods don't have to expose `validate`, I don't expect these methods to get called often, if at all, so it didn't seem worth the additional complexity.

Reply:
I agree. I think the main idea here is to have a way to skip the computationally expensive validation checks, so if a few simple validation checks remain, I don't perceive that as a big issue. If we identify these remaining checks as performance bottlenecks later in the year when we do profiling work, we can work to improve/resolve then.